src/Core/src/Platform/Android/Resources/values/styles.xml

```diff
@@ -28,9 +28,20 @@
     <style name="AppTheme" parent="Maui.MainTheme" />
     <style name="AppTheme.NoActionBar" parent="Maui.MainTheme.NoActionBar" />
 
-    <style name="MauiMaterialButton" parent="Widget.MaterialComponents.Button.Icon">
+    <style name="MauiMaterialButton" parent="Widget.MaterialComponents.Button">
+      <!-- remove all the min sizes as MAUI manages it -->
+      <item name="android:minWidth">0dp</item>
+      <item name="android:minHeight">0dp</item>
+      <!-- padding is the space between the edge of the background and the contents  -->
+      <item name="android:paddingLeft">0dp</item>
+      <item name="android:paddingRight">0dp</item>
+      <item name="android:paddingTop">0dp</item>
+      <item name="android:paddingBottom">0dp</item>
```

Comment:
We have to set the padding to 0 here otherwise the material button will try be smart and add all sorts of padding to our padding to try and get the perfect button.